conans/client/proxy.py

```diff
@@ -85,24 +86,35 @@ def _refresh():
             read_manifest, expected_manifest = self._paths.conan_manifests(conan_reference)
             if read_manifest.file_sums != expected_manifest.file_sums:
                 output.warn("Bad conanfile detected! Removing export directory... ")
-                _refresh()         
+                _refresh()
             else:  # Check for updates
-                try:  # get_conan_digest can fail, not in server
-                    upstream_manifest = self.get_conan_digest(conan_reference)
-                    if upstream_manifest.file_sums != read_manifest.file_sums:
-                        if upstream_manifest.time > read_manifest.time:
-                            output.warn("Current conanfile is older than remote upstream one")
-                            if self._update:
-                                _refresh()
-                        else:
-                            output.warn("Current conanfile is newer than remote upstream one")
-                except ConanException:
-                    pass
+                if self._check_updates:
+                    ret = self.update_available(conan_reference)
+                    if ret == 1:
+                        output.warn("There is a new conanfile upstream")
+                        if self._update:
+                            _refresh()
+                    elif ret == -1:
+                        output.warn("Current conanfile is newer than remote upstream one")
         else:
             output.info("Conanfile not found, retrieving from server")
             self._retrieve_conanfile(conan_reference, output)
         return conanfile_path
 
+    def update_available(self, conan_reference):
+        """Returns 0 if the conanfiles are equal, 1 if there is an update and -1 if 
+        the local is newer than the remote"""
+        if not conan_reference:
+            return 0
+        read_manifest, _ = self._paths.conan_manifests(conan_reference)
```

Comment:
I think this function shouldn't be called, as it is already called in get_conanfile(), maybe add read_manifest as optional parameter to avoid read in this case.